plugins/generator-1.16.5/forge-1.16.5/templates/item.java.ftl

```diff
@@ -329,6 +348,38 @@ import net.minecraft.entity.ai.attributes.Attributes;
         }
         </#if>
 
+        <#if hasProcedure(data.onFinishUsingItem) || data.hasEatResultItem()>
+            @Override public ItemStack onItemUseFinish(ItemStack itemstack, World world, LivingEntity entity) {
+        	    ItemStack retval =
+        		    <#if data.eatResultItem?? && !data.eatResultItem.isEmpty()>

```

Comment:
This should be data.hasEatResultItem() also, same in 1.18.x, otherwise if you select result item and disable food, this will still generate.

Please, test the code and check how it generates.

Reply:
> This should be data.hasEatResultItem() also, same in 1.18.x, otherwise if you select result item and disable food, this will still generate.
> 
> Please, test the code and check how it generates.

I tested, but I did not test the case a procedure could be selected and not the food checkbox.